File: DSA/SakshamDwivedi_590013147/Day18/D18P2.c
#include <stdio.h>

int arr[100];
int size;
int sw_key;
int j;

int main(){
    printf("Enter the size of the array (Less than 100) : ");
    scanf("%d", &size);
    printf("Enter the elements : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }

    for(int i = 1; i < size; i++){
        sw_key = arr[i];
        j = i - 1;
        while((j >= 0) && (arr[j] > sw_key)){
            arr[j+1] = arr[j];
            j -= 1;
        }
        arr[j+1] = sw_key;
    }
    printf("Sorted Array : ");
    for(int i = 0; i < size; i++){
        printf("%d ", arr[i]);
    }
}

File: DSA/SakshamDwivedi_590013147/Day18/D18P2.py
#Insertion sort
def insertion_sort(arr):
    for i in range(1, len(arr)):
        sw_key = arr[i]
        j = i-1
        while j >= 0 and arr[j] > sw_key:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = sw_key
    return arr

#Fixed Input
print(insertion_sort([3, 2, 6, 1, 7]))

#Custom Input
arr = list(int(x) for x in input("Enter the array : ").split())
print(insertion_sort(arr))
